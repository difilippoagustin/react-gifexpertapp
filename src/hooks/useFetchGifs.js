import { useEffect, useState } from "react";
import { GetGif } from "../helpers/GetGif";
export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    isLoading: true,
  });

  useEffect(() => {
    GetGif(category).then((res) => {
      setState({
        data: res,
        isLoading: false,
      });
    });
  }, [category]);

  return state;
};
