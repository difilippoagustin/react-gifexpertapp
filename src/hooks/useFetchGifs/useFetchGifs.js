import { useEffect, useState } from "react";
import PropTypes from "prop-types";

import { getGif } from "../../helpers/getGif/getGif";
export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    isLoading: true,
  });

  useEffect(() => {
    getGif(category).then((res) => {
      setState({
        data: res,
        isLoading: false,
      });
    });
  }, [category]);

  return state;
};
useFetchGifs.propTypes = {
  category: PropTypes.string.isRequired,
};
