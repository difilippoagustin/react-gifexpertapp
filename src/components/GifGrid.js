import React from "react";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export default function GifGrid({ category }) {
  const { data: images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeInDown">{category}</h3>
      {isLoading && "Loading data..."}
      <div className="card-grid">
        {images.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
}
