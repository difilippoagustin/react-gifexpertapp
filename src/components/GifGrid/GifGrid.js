import React from "react";
import PropTypes from "prop-types";
import { GifGridItem } from "../GifGridItem/GifGridItem";
import { useFetchGifs } from "../../hooks/useFetchGifs/useFetchGifs";

export default function GifGrid({ category }) {
  const { data: images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeInDown">{category}</h3>
      {isLoading && <p>"Loading data..."</p>}
      <div className="card-grid">
        {images.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
}
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
