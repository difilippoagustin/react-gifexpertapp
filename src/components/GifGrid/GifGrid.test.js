import React from "react";
import "@testing-library/jest-dom";

import { shallow } from "enzyme";
import GifGrid from "./GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs/useFetchGifs";

jest.mock("../../hooks/useFetchGifs/useFetchGifs");
describe("<GifGrid />", () => {
  const category = "category test";

  test("should display correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      isLoading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("should display images and should not display loading text when the data has been loaded", () => {
    const mockData = [
      { url: "https://image1.com", title: "Image 1", id: "1" },
      { url: "https://image2.com", title: "Image 2", id: "2" },
      { url: "https://image3.com", title: "Image 3", id: "3" },
    ];
    useFetchGifs.mockReturnValue({
      data: mockData,
      isLoading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    const loading = wrapper.find("p");
    expect(loading.exists()).toBeFalsy();
    expect(wrapper.find("GifGridItem").length).toBe(mockData.length);
  });
});
