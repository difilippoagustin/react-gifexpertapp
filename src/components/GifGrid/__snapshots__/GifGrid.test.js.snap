// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> should display correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    category test
  </h3>
  <p>
    "Loading data..."
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
